internal/provider/resource_dashboard.go

```diff
@@ -291,3 +362,64 @@ func (r *dashboardResource) Delete(ctx context.Context, req resource.DeleteReque
 func (r *dashboardResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
+
+func (r *dashboardResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	tflog.Info(ctx, "ModifyPlan called for dashboard resource")
+	if req.State.Raw.IsNull() || req.Plan.Raw.IsNull() {
+		tflog.Debug(ctx, "ModifyPlan: Skipping for new or destroyed dashboard resource")
+		return
+	}
+
+	var plan dashboardResourceModel
+	var state dashboardResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, "ModifyPlan: Checking for changes", map[string]interface{}{
+		"plan_preset_len":  len(plan.Preset.ValueString()),
+		"state_preset_len": len(state.Preset.ValueString()),
+		"presets_equal":    plan.Preset.Equal(state.Preset),
+	})
+
+	// Check preset for semantic changes
+	if !plan.Preset.IsNull() && !plan.Preset.IsUnknown() && !state.Preset.IsNull() && !state.Preset.IsUnknown() {
```

Comment:
please remove some indentation levels from this function with `if !; return` to make it more readable

for example, check here:
```
f plan.Preset.IsNull() || plan.Preset.IsUnknown() || state.Preset.IsNull() || state.Preset.IsUnknown() {
  return
}
```

same for `if plannedPreset != statePreset`